src/lib/comms/include/sol-connman.h

```diff
@@ -0,0 +1,382 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <sol-network.h>
+#include <sol-buffer.h>
+#include <sol-vector.h>
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @struct sol_connman_service
+ *
+ * @brief service struct
+ */
+struct sol_connman_service;
+
+/**
+ * @struct sol_connman_network_params
+ *
+ * @brief network params
+ *
+ * This struct contains the information of a network.
+ * it has the addr of network link addr, the network of
+ * netmask and its gateway of network.
+ */
+struct sol_connman_network_params {
+    /**
+     * @brief The network devices address
+     */
+    struct sol_network_link_addr addr;
+    /**
+     * @brief The network devices netmask
+     */
+    struct sol_network_link_addr netmask;
+    /**
+     * @brief The network gateway
+     */
+    struct sol_network_link_addr gateway;
+};
+
+/**
+ * @enum sol_connman_service_state
+ *
+ * @brief service state
+ *
+ * One of these must be chosen to set the state of service,
+ * with sol_connman_service_get_state()
+ *
+ * @note SOL_CONNMAN_SERVICE_STATE_REMOVE is used to show
+ * the service has been removed. When it is notified,
+ * the service has been removed in the system.
+ */
+enum sol_connman_service_state {
+    SOL_CONNMAN_SERVICE_STATE_UNKNOWN = 0,
```

Comment:
add the documentation about each state using the format:

```c
enum sol_connman_service_state {
     /**
      * @brief the service is idle (what does this mean?)
      *
      * a service is idle when situation X, Y and Z occur.
      */
```

Provide details on the meaning, like `REMOVE` should mention that the service handle (pointer) will become invalid.